admin-dev/themes/new-theme/js/pages/product/edit/virtual-product-manager.ts

```diff
@@ -76,4 +77,27 @@ export default class VirtualProductManager {
   private showContent(): void {
     this.$fileContentContainer.removeClass('d-none');
   }
+
+  private fillVirtualProductNameField(): void {
+    $(ProductMap.virtualProduct.fileUploadField).on('change', () => {
+      let fullPath = $(ProductMap.virtualProduct.fileUploadField).val()!;
```

Comment:
What's the use of the ending `!` ?

Reply:
I removed it 👍 

it's used in typescript to say that the value will never be null or undefined, I was trying something and forgot to remove it.... because `.val()` can actually be undefined if the field doesn't exist, so I checked for undefined below instead, it's better this way I guess